app/dashboard/date-picker/custom-display.tsx

```diff
@@ -0,0 +1,78 @@
+import { twentyFourHoursInMilliseconds } from '@/app/utils/constants';
+import StepBtn from './step-btn';
+
+interface CustomDisplayProps {
+  date: Date;
+  pickerRef: React.Ref<HTMLButtonElement>;
+  startDate: Date;
+  minDate: Date;
+  maxDate: Date;
+  onClick?: React.MouseEventHandler<HTMLButtonElement>;
+  onChange: (date: Date | null) => void;
+}
+
+const CustomDisplay = ({
+  date,
+  pickerRef,
+  startDate,
+  minDate,
+  maxDate,
+  onClick,
+  onChange,
+}: CustomDisplayProps) => {
+  const now = new Date();
+
+  const getReadableDate = (date: Date): string | null => {
+    const nowInDay = new Date(now.toDateString()).getTime();
+    const dateInDay = new Date(date.toDateString()).getTime();
+    switch (dateInDay) {
+      case nowInDay:
+        return 'Today';
+      case nowInDay + twentyFourHoursInMilliseconds:
+        return 'Tomorrow';
+      case nowInDay - twentyFourHoursInMilliseconds:
+        return 'Yesterday';
+      default:
+        return null;
```

Comment:
Optional: can use an empty string `''` for the `default` case to make the return type more consistent while keeping the distinguishable nature of this case.